specs/RainbowBridge/README.md

```diff
@@ -0,0 +1,388 @@
+|                               |                 |                               |
+|  .-------------.              |                 |              .-------------.  |
+|  | .-------------.            |                 |            .-------------. |  |
+|  '-| .-------------.          |                 |          .-------------. |-'  |
+|    '-|  contracts  |      (A) |                 | (1)      |  contracts  |-'    |
+|      '-----+-+-----'          |                 |          '-----+-+-----'      |
+|            | |                |                 |                | |            |
+|            \ /                |                 |                \ /            |
+|  +----------------------+     |                 |     +----------------------+  |
+|  |                      |     |                 |     |                      |  |
+|  |   EthEventVerifier   | (B) |                 | (2) | NearTxResultVerifier |  |
+|  |                      |     |                 |     |                      |  |
+|  +-------------+-+------+     |                 |     +-----+-+--------------+  |
+|                | |            |                 |           | |                 |
+|                \ /            |                 |           \ /                 |
+|          +--------------+     \                 |     +--------------+          |
+|          |              |      o - NearRelay --------->              |          |
+|          |              |     /                 |     |              |          |
+|          |  EthClient   | (C) |                 | (3) |  NearClient  |          |
+|          |              |     |                 /     |              |          |
+|          |              <--------- EthRelay  - o      |              |          |
+|          +--------------+     |                 \     +--------------+          |
+|                               |                 |                               |
++-------------------------------+                 +-------------------------------+
+```
+
+
+### Use Cases
+
+Please refer to the diagram above where (A), (B) and (C) appear to the left and (1), (2) and (3) appear the right.
+
+#### Use Case A-B-C-B-A 
+
+*NEAR smart contract must verify an event was recorded on the Ethereum blockchain*
+
+- A contract running on the *NEAR blockchain* requires verifiable proof that an event has been recorded on the *Ethereum blockchain*.  
+- The contract `(A)` invokes a method on the `EthEventVerifier` `(B)`, passing in the appropriate parameters to represent the event
+- `EthEventVerifier` `(B)` unpacks the Ethereum event and extracts the related block header which it passes on to the `EthClient` `(C)`
+- `EthClient` `(C)` returns a boolean indicating whether the block header was indeed recorded on the Ethereum blockchain
+- `EthEventVerifier` `(B)` passes this result on to the contract `(A)` as proof or rejection of the event as valid
```

Comment:
Rejection?

Reply:
i mean that the event is valid or invalid, proof it is valid or rejection of the claim that it is valid